Added suport for VS 2019 and CMake 3.14.0

ladland opened the pull request:
Updated CMake files with support for VS 2019.

Currently for packages to compile on a x64 based system the triplet (x64-windows) needs to be manually specified

ladland committed:
Added suport for VS 2019 and CMake 3.14.0

msftclas commented:
[![CLA assistant check](https://cla.opensource.microsoft.com/pull/badge/not_signed)](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5700) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5700) before we can accept your contribution.<br/><br/>:x: ladland [sign now](https://cla.opensource.microsoft.com/Microsoft/vcpkg?pullRequest=5700)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.opensource.microsoft.com/check/Microsoft/vcpkg?pullRequest=5700) it.</sub>

ladland committed:
Update vcpkg_configure_cmake.cmake

Fixed error in vcpkg_configure_cmake.cmake caused by new CMake generator naming scheme

ladland committed:
Update vcpkg_configure_cmake.cmake

Fixed error in vcpkg_configure_cmake.cmake caused by new CMake generator naming scheme

Rastaban commented:
Similar to #5603

Rastaban commented:
I am running additional tests on this PR.

Rastaban commented:
These changes were combined with #5702 and merged.  Thank you for the PR!
